src/components/Form/Widgets/PasswordWidget.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+	Box,
+	FormControl,
+	IconButton,
+	InputAdornment,
+	TextField,
+	Typography,
+} from '@mui/material';
+import * as React from 'react';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { WidgetProps } from '@rjsf/utils';
+import zxcvbn from 'zxcvbn';
+
+const STRENGTH_TITLES = ['Very weak', 'Weak', 'Good', 'Strong', 'Very strong'];
+const STRENGTH_STYLES = [
+	{ width: 0 },
+	{ width: '25%', backgroundColor: 'orange' },
+	{ width: '50%', backgroundColor: 'yellow' },
+	{ width: '75%', backgroundColor: 'green' },
+	{ width: '100%', backgroundColor: 'darkgreen' },
+];
+
+export interface PasswordStrengthProps {
+	password?: string;
+}
+
+const PasswordStrength = ({ password }: PasswordStrengthProps) => {
+	const [strengthScore, setStrengthScore] = React.useState<
+		number | undefined
+	>();
+
+	React.useEffect(() => {
+		if (password) {
+			try {
+				const { score } = zxcvbn(password);
+				setStrengthScore(score);
+			} catch {
+				// Ignore any errors, as we only want to show the strength meter if it is available.
+			}
+		}
+	}, [password]);
+
+	if (!password || strengthScore === undefined) {
+		return null;
+	}
+
+	return (
+		<Box>
+			<Typography>
+				Password strength: <em>{STRENGTH_TITLES[strengthScore]}</em>
```

Comment:
Could/should we use a Typography for the `em` as well to ensure the theme always applies to the text? (Not sure if it's necessary, just raising this in case it is. Let me know)

Reply:
I would keep using native elements instead of having a component that has to declare a font familly as we discussed in the call.